shell/common/display.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_COMMON_DISPLAY_H_
+#define FLUTTER_SHELL_COMMON_DISPLAY_H_
+
+namespace flutter {
+
+/// Unique IO per display that is stable until the Flutter application restarts.
+/// See also: `flutter::Display`
+typedef uint64_t DisplayId;
+
+/// Some devices do not support multiple displays, in such cases the embedding
+/// might not provide a device identifier. This value is used as the device id
+/// in such cases. There must only be one screen in such cases.
+static constexpr DisplayId kSingleScreenDeviceId = 0;
+
+/// To be used when the display refresh rate is unknown. This is in frames per
+/// second.
+static constexpr double kUnknownDisplayRefreshRate = 0;
+
+/// Display refers to a graphics hardware system consisting of a framebuffer,
+/// typically a monitor or a screen. This class holds the various display
+/// settings.
+class Display {
+ public:
+  Display(DisplayId display_id, double refresh_rate)
+      : display_id_(display_id), refresh_rate_(refresh_rate) {}
```

Comment:
Is it legal to create a `Display` with an "unknown" refresh rate or ID, or is that reserved for devices that don't have multiple screen capability?

Reply:
I think it should be ok to create a display with unknown refresh rate. I'm not sure when this could be the case, but say a platform call to get this info fails, given that this value is currently only advisory, I think we should allow that.

As it stands I'm hoping that all platforms can provide a display id, except for maybe single screen ones where we can use the value `kSingleScreenDeviceId`. So there shouldn't be a unknown ID in multiple devices case.

Reply:
I guess what I'm asking is: what if they provide zero, but that's a valid ID that they got from the platform?

Reply:
Oh, https://github.com/flutter/engine/pull/21355#discussion_r493910485 was partly answering this.

If `0` is a valid platform display id, the expectation is for the embedder to substitute it with another value and map it back to `0` if the need arises. I was basing this on [macOS](https://github.com/phracker/MacOSX-SDKs/blob/master/MacOSX10.9.sdk/System/Library/Frameworks/CoreGraphics.framework/Versions/A/Headers/CGDirectDisplay.h#L19) where `0` corresponds to null/invalid display.